vector/src/main/java/im/vector/riotx/features/home/room/detail/timeline/action/MessageActionsViewModel.kt

```diff
@@ -200,6 +206,27 @@ class MessageActionsViewModel @AssistedInject constructor(@Assisted
         } ?: ""
     }
 
+    private fun getRedactionReason(timelineEvent: TimelineEvent) =
+            (timelineEvent
+                    .root
+                    .unsignedData
+                    ?.redactedEvent
+                    ?.content
+                    ?.get("reason") as? String)
+                    ?.takeIf { it.isNotBlank() }
+                    ?.let { reason ->
+                        when (timelineEvent.root.senderId == timelineEvent.root.unsignedData?.redactedEvent?.senderId) {
```

Comment:
There is a method `Event.isRedacted()`, you could create another one below in the Event class `fun isRedactedBySameUser() = senderId == unsignedData?.redactedEvent?.senderId` on the same principle for a cleaner code